src/main/java/mate/academy/bookstore/controller/OrderController.java

```diff
@@ -0,0 +1,84 @@
+package mate.academy.bookstore.controller;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import mate.academy.bookstore.dto.order.CreateOrderRequestDto;
+import mate.academy.bookstore.dto.order.OrderResponseDto;
+import mate.academy.bookstore.dto.order.UpdateOrderRequestDto;
+import mate.academy.bookstore.dto.orderitem.OrderItemResponseDto;
+import mate.academy.bookstore.model.User;
+import mate.academy.bookstore.serivce.order.OrderService;
+import mate.academy.bookstore.serivce.orderitem.OrderItemService;
+import org.springframework.data.domain.Pageable;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.core.Authentication;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Order management", description = "Endpoints for managing orders")
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/orders")
+public class OrderController {
+    private final OrderService orderService;
+    private final OrderItemService orderItemService;
+
+    @PreAuthorize("hasRole('USER')")
+    @GetMapping
+    @Operation(summary = "Get all orders",
+            description = "Get all orders of the user")
+    public List<OrderResponseDto> getAllOrders(Pageable pageable, Authentication authentication) {
+        User user = (User) authentication.getPrincipal();
+        return orderService.getAllOrders(pageable, user);
+    }
+
+    @PreAuthorize("hasRole('USER')")
+    @GetMapping("/{orderId}/items")
+    @Operation(summary = "Get all order items for an order",
+            description = "Get all order items by order ID")
+    public List<OrderItemResponseDto> getOrderItems(@PathVariable Long orderId) {
+        return orderItemService.getAllOrderItems(orderId);
+    }
+
+    @PreAuthorize("hasRole('USER')")
+    @GetMapping("/{orderId}/items/{itemId}")
+    @Operation(summary = "Get order item by ID",
+            description = "Get order item by order ID and item ID")
+    public OrderItemResponseDto getOrderItem(
+            @PathVariable Long orderId,
+            @PathVariable Long itemId) {
+
+        return orderItemService.getOrderItem(orderId, itemId);
+    }
```

Comment:
Let's add a validation to service methods that the order that we have by id, belongs to the user who is making the request and not some other user, to be more secure